ocw/lib/EC2.py

```diff
@@ -89,45 +88,33 @@ def all_clusters(self):
-    def needs_to_delete_snapshot(snapshot, cleanup_ec2_max_snapshot_age_days) -> bool:
-        delete_older_than = date.today() - timedelta(days=cleanup_ec2_max_snapshot_age_days)
-        if datetime.date(snapshot['StartTime']) < delete_older_than:
-            regexes = [
-                re.compile(r'''^OpenQA upload image$'''),
-                re.compile(r'''^Created by CreateImage\([\w-]+\) for ami-\w+ from vol-\w+$''')
-            ]
-            for regex in regexes:
-                m = re.match(regex, snapshot['Description'].strip())
-                if m:
-                    return True
-        return False
+    def needs_to_delete_by_age(creation_time, cleanup_ec2_max_age_days) -> bool:
+        return datetime.date(creation_time) < (date.today() - timedelta(days=cleanup_ec2_max_age_days))
 
     def cleanup_snapshots(self, cleanup_ec2_max_snapshot_age_days):
         for region in self.all_regions:
             response = self.ec2_client(region).describe_snapshots(OwnerIds=['self'])
-            response['Snapshots'].sort(key=lambda snapshot: snapshot['StartTime'].timestamp())
             for snapshot in response['Snapshots']:
-                if EC2.needs_to_delete_snapshot(snapshot, cleanup_ec2_max_snapshot_age_days):
-                    self.log_info("Deleting snapshot {} in region {} with StartTime={}", snapshot['SnapshotId'],
-                                  region, snapshot['StartTime'])
+                if EC2.needs_to_delete_by_age(snapshot['StartTime'], cleanup_ec2_max_snapshot_age_days):
                     try:
                         if self.dry_run:
                             self.log_info("Snapshot deletion of {} skipped due to dry run mode",
                                           snapshot['SnapshotId'])
                         else:
+                            self.log_info("Deleting snapshot {} in region {} with StartTime={}",
+                                          snapshot['SnapshotId'], region, snapshot['StartTime'])
                             self.ec2_client(region).delete_snapshot(SnapshotId=snapshot['SnapshotId'])
                     except ClientError as ex:
                         if ex.response['Error']['Code'] == 'InvalidSnapshot.InUse':
                             self.log_info(ex.response['Error']['Message'])
                         else:
                             raise ex
 
-    def cleanup_volumes(self, cleanup_ec2_max_volumes_age_days):
-        delete_older_than = date.today() - timedelta(days=cleanup_ec2_max_volumes_age_days)
+    def cleanup_volumes(self, cleanup_ec2_max_age_days):
         for region in self.all_regions:
             response = self.ec2_client(region).describe_volumes()
             for volume in response['Volumes']:
-                if datetime.date(volume['CreateTime']) < delete_older_than:
+                if EC2.needs_to_delete_by_age(volume['CreateTime'], cleanup_ec2_max_age_days):
                     if self.volume_protected(volume):
```

Comment:
I guess this can be deleted.

Reply:
after thinking more about that , I decided to keep it . It will basically drop ability to protect volumes which are not attached to running VM . I renamed tag to be "pcw_ignore" to align this with all other places where we have similar logic . If you want to insist on deletion let's have separate chat about this

Reply:
"It will basically drop ability to protect volumes which are not attached to running VM" - I have no idea what you are trying to say with this sentence. What does `It` mean? What does `protect` means in this context?

Reply:
if you need to some volume not to be deleted forever , currently you can label it with "DO_NOT_DELETE" and than PCW will exclude it from its queries , so volume will be *protected* from PCW which otherwise will delete it after some time . If we simply drop this function together with it we will drop mechanism allowing to protect volumes from PCW

Reply:
Now I understand and now it also makes sense. Thanks, yes let's keep it! Marking as resolved.